consensus/beacon/consensus.go

```diff
@@ -192,6 +195,12 @@ func (beacon *Beacon) verifyHeader(chain consensus.ChainHeaderReader, header, pa
 	if header.UncleHash != types.EmptyUncleHash {
 		return errInvalidUncleHash
 	}
+	if header.Time > uint64(time.Now().Unix()+allowedFutureBlockTimeSeconds) {
```

Comment:
Why do we need to check the timestamp here? It's passed by CL and we should always assume CL will verify it, like it's not older than parent. 

Regarding the future block, it's indeed possible that the given timestamp is in future because CL and EL can in different machine. But does it really matter? And why do we have this 1 second allowance?

Reply:
1) Yes that was how the spec used to be. Now it changed a bit and left this underspecified. During `engine_executePayload` we check that the timestamp matches, however we don't do it for historical blocks passed by sync.

2) The 1 second allowance is exactly for that, if two machines are on slightly different clocks. Again, the CL should check it, but I think it is better to not rely on the CL for that

Reply:
So what's the side effect for receiving the "invalid timestamp"?

- if the timestamp is even older than parent, then a consensus error is returned and the given block is rejected(looks correct)
- if the timestamp is in the future, the block is supposed to be added into the future queue. But PoS blocks are not allowed caching in the future queue. So eventually the block is ignored silently? (also looks correct). And next time the block sync will fill the gap I think.

Reply:
I don't get this change. I'm perfectly fine with 
```
if header.Time <= parent.Time {
		return errOlderBlockTime
	}
```
But why do we also require the child block to be _exactly_ `parent+1`?

Reply:
Also, traditionally, we've always saved `ErrFutureBlock` for last, so that it basically means "everything checks out, _except_ the timestamp is a bit off". We should not change that logic, so please move it last